core/jsAndWasmShared/src/LocalDateTime.kt

```diff
@@ -25,34 +27,36 @@ public actual class LocalDateTime internal constructor(internal val value: jtLoc
     public actual constructor(date: LocalDate, time: LocalTime) :
             this(jtLocalDateTime.of(date.value, time.value))
 
-    public actual val year: Int get() = value.year().toInt()
-    public actual val monthNumber: Int get() = value.monthValue().toInt()
+    public actual val year: Int get() = value.year()
+    public actual val monthNumber: Int get() = value.monthValue()
```

Comment:
Why are some casts added and some removed?

Reply:
The js-joda external declarations were updated - now there is precise parameter and return types were specified  (instead of `Number`).  It is done because `Wasm` do not have interop for `Number`. This change turned out that some of casts are become redundant and some casts became necessary.

Reply:
Here I made half of work to separate Joda API from external declarations. This code will be doubled to provide support for the Wasm.
https://github.com/Kotlin/kotlinx-datetime/commit/e78597df98eb0a0aade70f43e52c40299ce5ddd8